add missing countries

waltermichelraja opened the pull request:
# Pull Request
## Description
This PR adds and updates country-level solar capacity data and improves the country forecast page.

**Changes included:**

1. **`get_solar_capacities.py`**
   - Utility script to fetch solar capacity by country code.
   - Handles missing countries, edge cases, and country name mappings.
   - Validates dataset and warns if a country is missing.
   - Updated capacities with latest verified data sources.
   - Ensured proper ISO alpha-3 country codes for all entries. [Lesotho LSO had a redundant row with LES as ISO code, so LES was removed]

2. **`country.py` (country page)**
   - Updated `get_country_coords()` to include small states not in geojson.
   - Improved timezone handling with `get_country_timezone()`.
   - Forecast display now uses local time robustly.
   - Integrated solar capacities into forecast calculations.

### Fixes: #51 

## How Has This Been Tested?

- Verified CSV loads correctly and contains all countries.
- Tested `get_solar_capacity()` for multiple countries, including edge cases (small countries and missing entries).
- Checked country page in Streamlit:
  - Forecasts render correctly for selected countries.
  - Timezones and local time conversions are accurate.
  - Fallback coordinates work for microstates.
- [x] Tested Streamlit country page forecasts.

## Checklist:

- [x] Code follows [OCF coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] Self-review of code completed
- [ ] Corresponding changes to documentation (`README.md`) added
- [ ] No spelling errors in code or documentation
- [x] Tests and sanity checks for new functionality performed

waltermichelraja commented:
@peterdudfield 
<img width="1027" height="698" alt="Screenshot from 2025-09-27 08-06-40" src="https://github.com/user-attachments/assets/ea63113a-ba4f-43f8-8c52-26e621a8bd62" />

I guess now everything is covered on the map...

waltermichelraja commented on src/v1/data/get_solar_capacities.py:
```diff
@@ -41,17 +41,17 @@
     "TJK": {
```
i haven't removed countries, i added the ones that were not present and resolved for cases where countries had duplicate/invalid country codes...
you can test them though...

peterdudfield reviewed (changes requested):
Im worried about al the changes in the countries. 
Its unclear why some countries are removed?

waltermichelraja commented:
@peterdudfield 

i have gone through your reviews,

- countries like for example like Malta already present in ember dataset, so i removed them to prevent redundancy [I'm sure that I haven't missed out any countries although i have removed some]
- Lesotho - had two had 2 fields with different country code so i had to remove the incorrect one and keep the one with valid country code
- some countries are too small to be seen on map scale so i had to create fallback_coords in country.py

let me know if this will work

waltermichelraja commented:
> ok, sounds good, I wonder if in a differen PR we should order the csv by country name. Then we would be able to see changes in capacity a little more clearer

alright!

waltermichelraja commented:
> @all-contributors please add @waltermichelraja for code

thank you! @peterdudfield
